src/rpc/handlers/AMMInfo.cpp

```diff
@@ -0,0 +1,207 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <data/DBHelpers.h>
+#include <rpc/AMMHelpers.h>
+#include <rpc/handlers/AMMInfo.h>
+
+#include <ripple/protocol/AMMCore.h>
+
+namespace {
+
+std::string
+toIso8601(ripple::NetClock::time_point tp)
+{
+    using namespace std::chrono;
+    static auto constexpr rippleEpochOffset = seconds{rippleEpochStart};
+
+    return date::format(
+        "%Y-%Om-%dT%H:%M:%OS%z",
+        date::sys_time<system_clock::duration>(system_clock::time_point{tp.time_since_epoch() + rippleEpochOffset})
+    );
+};
+
+}  // namespace
+
+namespace rpc {
+AMMInfoHandler::Result
+AMMInfoHandler::process(AMMInfoHandler::Input input, Context const& ctx) const
+{
+    using namespace ripple;
+
+    auto const range = sharedPtrBackend_->fetchLedgerRange();
+    auto const lgrInfoOrStatus = getLedgerInfoFromHashOrSeq(
+        *sharedPtrBackend_, ctx.yield, input.ledgerHash, input.ledgerIndex, range->maxSequence
+    );
+
+    if (auto const status = std::get_if<Status>(&lgrInfoOrStatus))
+        return Error{*status};
+
+    auto const lgrInfo = std::get<LedgerInfo>(lgrInfoOrStatus);
+
+    if (input.accountID) {
+        auto keylet = keylet::account(*input.accountID);
+        if (not sharedPtrBackend_->fetchLedgerObject(keylet.key, lgrInfo.seq, ctx.yield))
+            return Error{Status{RippledError::rpcACT_NOT_FOUND}};
+    }
+
+    auto ammKeylet = keylet::amm(input.issue1, input.issue2);
+    auto const ammBlob = sharedPtrBackend_->fetchLedgerObject(ammKeylet.key, lgrInfo.seq, ctx.yield);
+
+    if (not ammBlob)
+        return Error{Status{RippledError::rpcACT_NOT_FOUND}};
+
+    auto const amm = SLE{SerialIter{ammBlob->data(), ammBlob->size()}, ammKeylet.key};
+    auto const accID = amm.getAccountID(sfAccount);
```

Comment:
1 there are too many account here. I suggest to use more describable name like 
'lpAccount' means the account to check lp
'ammAccount' means the amm's account
2 I think we don't need to check the existence of amm account. It should exist if amm object exists? That being said, i am not against to check it just in case, but the error message should be more specific , like "amm object not found"? because the input might contain the other account, the error message may misleading.

Reply:
How does my new changes look like ?